src/routes/api/login/+server.ts

```diff
@@ -4,14 +4,19 @@ import { authenticateUser } from '$lib/server/user';
 
 export async function POST({ cookies, request }: RequestEvent) {
 	try {
-		const { userId, userName } = await request.json() as { userId: string; userName: string };
+		const { userId, userName, accessToken } = (await request.json()) as {
+			userId: string;
+			userName: string;
+			accessToken: string;
+		};
 		const record = await authenticateUser(userId, userName);
 		if (record.status === 'active') {
 			if (cookies.get('sessionid') === undefined) {
 				const session = await createSession(record);
 				let expires = new Date();
 				expires.setMonth(expires.getMonth() + 1);
 				cookies.set('sessionid', session.id, { path: '/', expires });
+				cookies.set('accesstoken', accessToken, { path: '/', expires });
```

Comment:
I think the accessToken is only valid for a couple hours after the user logins into FB, right?  Will that cause any issues?

Reply:
if the access token fails, then it will require the user to relogin.

Reply:
Ok I think this is a change in behavior from the current code, which only requires the user to log in through FB once per month.  This will require them to log in every 2 hours?  Seems like that could be a bit annoying for the user.

Reply:
we should test that actually. i just enabled https://superhome-scheduler-git-cleanup-auth-freedive-superhome.vercel.app/ in the facebook app login right now.

Reply:
that logged out. currently this uses only access token to retrieve the profile pic nothing else.

Reply:
Yes, really the only reason to use FB login is so that we don't have to deal directly with storing usernames and passwords, and to give the admins a way of associating the user with some kind of identification like an FB profile.  Also since Messenger is still used for some communications, it's nice to be able to easily associate the Messenger user with the corresponding app user.

Reply:
i think i should be able to generate a non-expiring profile image link somewhere. then this auto logout issue should resolved.

Reply:
this is addressed